tests/integ/sagemaker/workflow/test_model_steps.py

```diff
@@ -0,0 +1,1138 @@
+        drift_check_baselines=drift_check_baselines,
+        customer_metadata_properties=customer_metadata_properties,
+    )
+    step_model_register = ModelStep(
+        name="MyRegisterModelStep",
+        step_args=step_args,
+    )
+
+    pipeline = Pipeline(
+        name=pipeline_name,
+        parameters=[
+            model_uri_param,
+            metrics_uri_param,
+            instance_type,
+            instance_count,
+        ],
+        steps=[step_model_register],
+        sagemaker_session=pipeline_session,
+    )
+
+    try:
+        response = pipeline.create(role)
+        create_arn = response["PipelineArn"]
+
+        for _ in retries(
+            max_retry_count=5,
+            exception_message_prefix="Waiting for a successful execution of pipeline",
+            seconds_to_sleep=10,
+        ):
+            execution = pipeline.start(
+                parameters={"model_uri": model_uri, "metrics_uri": metrics_uri}
+            )
+            response = execution.describe()
+
+            assert response["PipelineArn"] == create_arn
+
+            try:
+                execution.wait(delay=30, max_attempts=60)
+            except WaiterError:
+                pass
+            execution_steps = execution.list_steps()
+
+            assert len(execution_steps) == 1
+            failure_reason = execution_steps[0].get("FailureReason", "")
+            if failure_reason != "":
+                logging.error(
+                    f"Pipeline execution failed with error: {failure_reason}." " Retrying.."
+                )
+                continue
+            assert execution_steps[0]["StepStatus"] == "Succeeded"
+
+            response = pipeline_session.sagemaker_client.describe_model_package(
+                ModelPackageName=execution_steps[0]["Metadata"]["RegisterModel"]["Arn"]
+            )
+
+            assert (
+                response["ModelMetrics"]["Explainability"]["Report"]["ContentType"]
+                == "application/json"
+            )
+            assert (
+                response["DriftCheckBaselines"]["Bias"]["PreTrainingConstraints"]["ContentType"]
+                == "application/json"
+            )
+            assert (
+                response["DriftCheckBaselines"]["Explainability"]["Constraints"]["ContentType"]
+                == "application/json"
+            )
+            assert (
+                response["DriftCheckBaselines"]["ModelQuality"]["Statistics"]["ContentType"]
+                == "application/json"
+            )
+            assert (
+                response["DriftCheckBaselines"]["ModelDataQuality"]["Statistics"]["ContentType"]
+                == "application/json"
+            )
+            assert response["CustomerMetadataProperties"] == customer_metadata_properties
+            break
+    finally:
+        try:
+            pipeline.delete()
+        except Exception as error:
+            logging.error(error)
+
+
+def test_model_registration_with_model_repack(
```

Comment:
test_pytorch_model_registration_creation_with_custom_inference

Move this up next to the first one. They form a pair.

Reply:
Updated in the new commit